### PR TITLE
Fixing de-autocast

### PR DESCRIPTION
**Pull request by borisfom**
Signed-off-by: Boris Fomitchev <bfomitchev@nvidia.com>

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/parts/submodules/multi_head_attention.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 import torch
 import torch.nn as nn
 
-from nemo.collections.common.parts.training_utils import avoid_float16_autocast_context
+from nemo.utils.export_utils import avoid_float16_autocast_context
```
Comment (titu1994) [inline nemo/collections/asr/parts/submodules/multi_head_attention.py:40]: This isnt export related context, should go into common training utils

File: nemo/collections/common/parts/training_utils.py [hunk 2]
```diff
@@ -1,49 +0,0 @@
-# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
```
Comment (titu1994) [inline nemo/collections/common/parts/training_utils.py:1]: Revert, these aren't export specific, but general training specific context managers

**Review by titu1994 — CHANGES_REQUESTED** [review]
Refactor to revert move of the context managers out of common training utils

File: nemo/collections/common/parts/training_utils.py (hunk 2 above)
Comment (borisfom) [reply nemo/collections/common/parts/training_utils.py]: Not training-specific either. Moved them to new cast utilities module.

**Review by titu1994 — APPROVED** [review]
Is it necessary to move qkv tensor to fp32? It will take 4x more memory for the score tensor then

**Comment by borisfom** [comment]
> Is it necessary to move qkv tensor to fp32? It will take 4x more memory for the score tensor then
Well I assumed that was the intent of the autocast-disable scope added.  Otherwise it has no effect I believe. 
We should definitely re-visit autocast policy around LayerNorm especially that I see more PRs around that. 
BTW, why 4x increase ? Should be 2x.

**Comment by titu1994** [comment]
Attention map is LxL so 2x2 increase in memory cost. Ok let's review the auto cast stuff

**Pull request merged**